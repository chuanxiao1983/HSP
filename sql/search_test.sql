SET client_min_messages TO WARNING;

CREATE EXTENSION IF NOT EXISTS hsp;
CREATE EXTENSION IF NOT EXISTS hstore;


CREATE TABLE test_hsp(id SERIAL PRIMARY KEY, val hmcode(64));

INSERT INTO test_hsp(val) VALUES ('{0,0,0,0,0,0,0,0}'),('{0,0,0,0,1,0,0,0}'),('{0,0,0,0,1,1,0,0}'),('{0,1,0,2,0,3,0,0}');

SELECT set_m(3);

SELECT create_index('test_hsp', 'val');

SELECT get_total();

SELECT * from test_hsp where hamming_distance('{0,0,0,0,0,0,0,0}', val) <= 1;

SELECT * from test_hsp order by hamming_distance('{0,0,0,0,0,0,0,0}', val) limit 2;