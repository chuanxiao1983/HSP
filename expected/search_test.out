SET client_min_messages TO WARNING;
CREATE EXTENSION IF NOT EXISTS hsp;
CREATE EXTENSION IF NOT EXISTS hstore;
CREATE TABLE test_hsp(id SERIAL PRIMARY KEY, val hmcode(64));
INSERT INTO test_hsp(val) VALUES ('{0,0,0,0,0,0,0,0}'),('{0,0,0,0,1,0,0,0}'),('{0,0,0,0,1,1,0,0}'),('{0,1,0,2,0,3,0,0}');
SELECT set_m(3);
 set_m 
-------
 
(1 row)

SELECT create_index('test_hsp', 'val');
 create_index 
--------------
 
(1 row)

SELECT get_total();
 get_total 
-----------
         4
(1 row)

SELECT * from test_hsp where hamming_distance('{0,0,0,0,0,0,0,0}', val) <= 1;
 id |        val        
----+-------------------
  1 | {0,0,0,0,0,0,0,0}
  2 | {0,0,0,0,1,0,0,0}
(2 rows)

SELECT * from test_hsp order by hamming_distance('{0,0,0,0,0,0,0,0}', val) limit 2;
 id |        val        
----+-------------------
  1 | {0,0,0,0,0,0,0,0}
  2 | {0,0,0,0,1,0,0,0}
(2 rows)

